### ISSUE TITLE
Assist (reply to) request for self-hosted Thimble instance.

### ISSUE THREAD

**Issue by flukeout** [mozilla/thimble.mozilla.org#1707]
Via email, we got a request for assistance with setting up a self-hosted instance from Davide.

@gideonthomas Is the newly created Vagrant set-up fully documented?

Davide, are you able to try the directions currently listed on the Readme.MD and let us know where the problems lie?

**Comment by humphd** [comment]
The docs at https://github.com/mozilla/thimble.mozilla.org#setupinstallation are up to date, and this will work for doing a self-hosted version.  It *is* intended more as a dev thing vs. a turn-key installation for production; however it will get things started.

I've also documented the setup process at the start of this post: http://blog.humphd.org/fixing-a-bug-in-mozilla-thimble/

If we find that people do want a turn-key, deployment style thing, we should file that as a separate bug and explore.  There are ways of doing that via packaging on Linux, macOS, etc.

**Comment by davenoah** [comment]
Hi, thanks for you replies, sorry but I heave read now the email.

The problems that I encountered are errors with npm install on id.webmaker.org.
I'm trying to install everything manually because I'm using an ec2 instance (but I tried a DigitalOcean droplet as well) and virtualbox+vagrant doesn't work. The instance is clean, I just installed build-essentials, libssl-dev, git and Node with NVM.

Example if I use Node 4.6 (that should be the version compatible with all the services):
- Bramble npm install interrupt with a grunt build browser but anyway it works when I start it
- Thimble works
- id.webmaker.org interrupt the npm install with an error on autoless, I tried to run the command twice and with other versions of Node but doesn't work and I can't proceed

Today I'll try with the vagrant setup in local.

**Comment by humphd** [comment]
cc @cadecairos.

I wonder if we should put some time into making this install-able via `apt-get` or the like, or also having things run with docker.  I bet some of my students could help with that, if there was consensus on how we want to approach it.

**Comment by Pomax** [comment]
this might also require doing a dep-check to see which dependencies have moved up one or more major versions. I know that autoless has been replaced by postcss these days, so there might be some maintenance work in addition to packaging.

**Comment by humphd** [comment]
Agreed.  We should be filing bugs on both Thimble + Brackets to update our deps.

**Comment by humphd** [comment]
Filed bugs: https://github.com/mozilla/thimble.mozilla.org/issues/1743 and https://github.com/mozilla/brackets/issues/589

**Comment by humphd** [comment]
This may or may not be a "good first bug," but I'd love to see one of my students try it.

**Comment by Pomax** [comment]
starting a bug and documenting what needs to be done, even if you end up going "but I'm not the person to be actually doing this" is an excellent open source contribution

**Comment by gideonthomas** [comment]
@davenoah would you be able to copy/paste the exact error you're getting with autoless in `id.webmaker.org`?

**Comment by humphd** [comment]
I spoke to one of my colleagues today, who specializes in packaging on Linux, and he suggests the right way to do this is probably to use Docker for each of the bits we need, and do self-hosted instances that way.  I know we're a bunch of the way there, but I wonder if anyone has the most recent info on this?  @cadecairos do you know?

**Comment by cadecairos** [comment]
I don't have much experience with dpkg, but I have been keeping my eye on two up-and-coming Linux app distribution methods that may or may not work for this (I'd have to dive into their developer documentation first) 

There's snaps by Canonical: https://snapcraft.io/
And flatpack from the FSF: http://flatpak.org/

A quick look at both did suggest that flatpack is more geared towards desktop app distribution.

**Comment by shubham0794x** [comment]
I would like to work on this bug. @humphd @gideonthomas

**Comment by humphd** [comment]
@shubham0794x if you want to, sure.  The right way to do this is likely to make Thimble and all of its dependencies work using Docker.  I've seen some work happening on this elsewhere in https://github.com/mgebundy/docker-thimble-nouser, which you might use/contribute to, etc.

**Comment by humphd** [comment]
Moving this out of the Assigned/GoodFirstBug labels, since it is neither.
